### PR TITLE
Formal internal review of Teams baseline

### PR DESCRIPTION
**Pull request by nanda-katikaneni**
#  #
## 🗣 Description ##
This PR is designed to facilitate a formal review of the revised Teams baseline from CISA SCuBA principals and the Scuba development team. Following this review the baseline should be in a state for sending to the CISA agency formal review process.

## 💭 Motivation and context ##
A broad review by both the development team SMEs and the CISA SCuBA principals ensures a high quality update in line with agency goals and consistent policy format and styling.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [ ] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [ ] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [ ] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

File: baselines/teams.md [hunk 1]
```diff
@@ -385,7 +385,7 @@ All the settings in this section are configured in the **Microsoft Teams admin c
 
 2.  Select **Teams** > **Teams Settings**.
 
-3.  Under the **Email integration** section, set **Allow users to send
+3.  Under the **Email integration** section, set **Users can send
```
Comment (gdasher) [inline baselines/teams.md:388]: For MS.TEAMS.2.3v1, the note about not being available in GCC is located with the policy.  For this one, it is in license requirements.  Suggest being consistent across the two.  I prefer the policy locaton as a note vs. license for this, reserving license for 3 vs. 5.

File: baselines/teams.md
Comment (gdasher) [inline baselines/teams.md:1]: for MS.TEAMS.5.1v1: suggest further clarifying how agency should pick specific apps in the microsoft category.  For example, should we include a baseline list?  or at least criteria for picking (E.g. something simple like its used in the agency)?  As written, I missed what should be done.  I think we need to be a little more clear here, even though we can't be prescriptive as to the exact list most likely.

**Review by gdasher — CHANGES_REQUESTED** [review]
A few minor changes requested.  Largely looks good.

**Comment by tkol2022** [comment]
### Spelling
"Specific" is misspelled as "specifc" a couple of times in the document.

**Comment by tkol2022** [comment]
### Section 5 policy update suggestions
I'm suggesting that we rephrase the section 5 policies related to Microsoft, third-party and custom apps so that the policy states that only apps "approved" by an agency can be installed. Right now we have kind of a mixed bag where policy 5.1 alludes to "specific" apps which indirectly means "agency approved" apps but policy 5.2 says that 3rd party/custom apps are outright disallowed (even though the instructions indicate that agency approved apps are okay). To standardize the policies here are some suggested new policy descriptions (feel free to tweak):

- MS.TEAMS.5.1v1 - Agencies SHOULD only allow installation of Microsoft Apps approved by the agency.
- MS.TEAMS.5.2v1 - Agencies SHOULD only allow installation of third-party or custom apps approved by the agency.

W/r/t to implementation, an agency would have a couple of configuration options and still be compliant with the policies:
- "Block all apps" (strictest setting available) 
- "Allow specific apps and block all others". If they select Allow specific apps the admin must then configure which apps are "approved" for use by the agency and can be installed by users.

**Comment by tkol2022** [comment]
### Section 5 Implementation section
I'm recommending the following updates to the instructions:

- MS.TEAMS.5.1v1 instructions - Add that "Block all apps" (strictest setting available) is also an acceptable option. We wouldn't want to flag an agency for being non compliant if their setting is at the strictest available option.
- MS.TEAMS.5.2v1 instructions - Now that you added the "Org-wide app settings" configuration buttons, we need to re-arrange the overall order of the instructions so that the "Org-wide app settings" are first since those take precedence and affect whether the "Global (Org-wide default)" settings are configurable or not. For example, if you set the "Org-wide app settings" configuration Third-party apps to Off, then you cannot configure the "Global (Org-wide default)" Third party apps (it is disabled in the admin portal). To rectify this we can structure the instructions this way:
  - Starting at step 4 we can say something like "If you are blocking all third-party apps, then Select Teams apps > Manage Apps. Select Org-wide app settings. Under Third-pary apps turn **Off** Third-pary apps." Then that is all they need to do. The same applies for custom apps.
  - For step 5 we would provide the instructions for setting up the "Allow specific apps..." option. "If you are configuring so that only approved third-party apps are allowed, then Select Teams apps > Manage Apps. Select Org-wide app settings. Under Third-pary apps turn **On** Third-pary apps." Then you would provide what is currently steps 3-6 for configuring only approved apps in the "Global (Org-wide default)" policy.
  - Again the main point or re-ordering the 5.2 instructions is so that the "Org-wide app settings" configuration buttons get configured first.
  - Step 9 contains a spelling error "Under Third-pary apps turn off Third-pary apps".

**Comment by tkol2022** [comment]
### Potential policy split
I'm wondering if it would be cleaner to split policy 5.2 into two separate policies, one for third party apps and one for custom apps? I believe these are really two distinct configuration items and I also think the Rego code might be cleaner if we split it up as per below:

- MS.TEAMS.5.2v1 - Agencies SHOULD only allow installation of third-party approved by the agency.
- MS.TEAMS.5.3v1 - Agencies SHOULD only allow installation of custom apps approved by the agency.

The Rego code for each of these would examine the "Org-wide app settings" configuration and potentially the Global (Org-wide default) setting if necessary. I'm not familiar with the current Rego code for these policies and I don't really have a strong opinion on this one, so go with your best judgement. My main concern is more streamlined code with potentially less logic errors in the long term.

**Comment by tkol2022** [comment]
### Nitpicking

- Some of the implementation instructions do not have periods at the end, whereas some do.
- Section 5 license requirements need a capital letter at the start of the sentence.

**Review by tkol2022 — APPROVED** [review]
I'm approving this PR with the assumption that my comments will be addressed. Thanks for listening to all the feedback.

File: baselines/teams.md
Comment (ahuynhMITRE) [reply baselines/teams.md:1]: updated to add more clarity that it is approved and used by the agency. Too many apps to propose a baseline list.

**Comment by ahuynhMITRE** [comment]
> 

updated and added "and used" to also address Grants comment above. Will create an issue to account for these options for the code once internal review is merged.

**Comment by ahuynhMITRE** [comment]
> 

updated!

File: baselines/teams.md
[the remark this answers is not in the document]
Comment (ahuynhMITRE) [reply baselines/teams.md:1]: updated

[the remark this answers is not in the document]
Comment (ahuynhMITRE) [reply baselines/teams.md:1]: updated section to repoint to safe attachments and also included license requirement if using defender

[the remark this answers is not in the document]
Comment (ahuynhMITRE) [reply baselines/teams.md:1]: decided to remove the policy also added the defender resource for safe links.

File: baselines/teams.md (hunk 1 above)
Comment (nanda-katikaneni) [reply baselines/teams.md:396]: Updated to make licensing and region applicability (GCC, GCC High, DoD) consistent across policies.

File: baselines/teams.md
[the remark this answers is not in the document]
Comment (nanda-katikaneni) [reply baselines/teams.md:1]: Fixed.

[the remark this answers is not in the document]
Comment (nanda-katikaneni) [reply baselines/teams.md:1]: Fixed

[the remark this answers is not in the document]
Comment (nanda-katikaneni) [reply baselines/teams.md:1]: Removed the redundancy and made references to GCC, GCC High and DoD consistent across policies.

[the remark this answers is not in the document]
Comment (nanda-katikaneni) [reply baselines/teams.md:1]: Fixed

[the remark this answers is not in the document]
Comment (nanda-katikaneni) [reply baselines/teams.md:1]: Updated.

[the remark this answers is not in the document]
Comment (nanda-katikaneni) [reply baselines/teams.md:1]: Fixed

[the remark this answers is not in the document]
Comment (nanda-katikaneni) [reply baselines/teams.md:1]: Fixed

[the remark this answers is not in the document]
Comment (nanda-katikaneni) [reply baselines/teams.md:1]: Rephrased rationale and updated license requirements.

**Pull request merged**